server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -108,6 +108,11 @@ public SearchResponse(SearchResponseSections internalResponse, String scrollId,
         assert skippedShards <= totalShards : "skipped: " + skippedShards + " total: " + totalShards;
     }
 
+    public SearchResponse(SearchResponse clone) {
+        this(clone.internalResponse, clone.scrollId, clone.totalShards, clone.successfulShards, clone.skippedShards,
+            clone.tookInMillis, clone.shardFailures, clone.clusters);
+    }
+
```

Comment:
I didn't want to overload the original class since it has its own definition of partial results (totalShards != successfulShards). The `partial` flag here is only relevant in the context of an async search.